core/server/worker/src/main/java/alluxio/worker/block/meta/StorageDir.java

```diff
@@ -125,7 +125,9 @@ private void initializeMeta() throws BlockAlreadyExistsException, IOException,
     }
     for (File path : paths) {
       if (!path.isFile()) {
-        LOG.error("{} in StorageDir is not a file", path.getAbsolutePath());
+        if (!path.getName().equals(ServerConfiguration.get(PropertyKey.WORKER_DATA_TMP_FOLDER))) {
```

Comment:
Is the intention to _only_ filter out this tmp folder, or any blocks within it? Because as it, won't this only filter out just that single folder?

Reply:
There should really only be one object in that directory that isn't a file which is the .tmp_blocks directory. Listing the objects underneath is not needed since this directory will get deleted anyways. The main point is to avoid logging the ERROR since it has/will alarm certain users.

Reply:
My last comment is how the value of the `WORKER_DATA_TMP_FOLDER` might match `path.getName()` - can the user specify a relative path? will it still match `path.getName()` if the config specifies a relative or absolute?

Reply:
WORKER_DATA_TMP_FOLDER must be a relative path. We don't allow an absolute path for this property. Path.getName() will also only get the last index of the path so this match should be correct.

Reply:
couldn't someone specify `path1/path2` - configuration would return that, but then the `path.getName()` would only have `path2`?

Reply:
I think path.getName() would only have path1, but I see your concern. Is this something we reasonably see happening?

Reply:
If the goal is to remove the message we need to account for the property being configured to something other than the default